numba_cuda/numba/cuda/dispatcher.py

```diff
@@ -535,8 +537,13 @@ def _prepare_args(self, ty, val, stream, retr, kernelargs):
         """
 
         # map the arguments using any extension you've registered
-        for extension in reversed(self.extensions):
-            ty, val = extension.prepare_args(ty, val, stream=stream, retr=retr)
+        if self.extensions:
+            for extension in reversed(self.extensions):
+                ty, val = extension.prepare_args(
+                    ty, val, stream=stream, retr=retr
+                )
+        elif handler := _arg_handlers.get(type(val)):
```

Comment:
I'm not sure the if / elif logic here is correct. It seems to implement "If there are any extensions registered for this kernel, ignore all globally-registered extensions". But I could imagine a scenario where there is a globally-registered extension for one argument, and a "locally"-registered one for another argument - I think this situation is precluded from working in the current implementation.